tls/s2n_psk.c

```diff
@@ -73,6 +73,32 @@ int s2n_psk_set_secret(struct s2n_psk *psk, const uint8_t *secret, uint16_t secr
     return S2N_SUCCESS;
 }
 
+S2N_RESULT s2n_psk_clone(struct s2n_psk *new_psk, struct s2n_psk *original_psk)
+{
+    if (original_psk == NULL) {
+        return S2N_RESULT_OK;
+    }
+    ENSURE_REF(new_psk);
+
+    struct s2n_psk psk_copy = *new_psk;
+
+    /* Copy all fields from the old_config EXCEPT the blobs, which we need to reallocate. */
+    *new_psk = *original_psk;
+    new_psk->identity = psk_copy.identity;
+    new_psk->secret = psk_copy.secret;
+    new_psk->early_secret = psk_copy.early_secret;
+    new_psk->early_data_config = psk_copy.early_data_config;
```

Comment:
Nit: We could explicitly set the  `type`, `hmac_alg` and `obfuscated_ticket_age` as well from original_psk here?

Reply:
We are copying those fields from original_psk-- that's what line 86 does. We don't have to call them out because they're just primitives and are handled by a shallow copy. We call out identity, secret, etc because we CAN'T shallow copy them-- we need to allocate new memory and do a deep copy.

Reply:
I understand its NOT necessary and it is absolutely acceptable to do a shallow copy, I was only suggesting to do a deep copy for all parameters in the nit above.

Reply:
There's no shallow/deep distinction for primitives.